### PR TITLE
Particles on scene change

### PR DESCRIPTION
**Pull request by AntonAsbrink**
Particles appear when using the teleporter

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Player.cpp [hunk 1]
```diff
@@ -7,6 +7,9 @@
 #include "Shop.h"
 #include "Events/EventBus.h"
 #include "ECS/SceneManager.h"
+#include "Misc/AssetLoader.h"
+#include "ECS/Components/TemporaryLifeComponent.h"
+#include "Renderer/Texture/Texture2DGUI.h"
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Player.cpp]: Why did u add these here?

File: StortSpelProjekt/Game/src/Gamefiles/Player.h [hunk 2]
```diff
@@ -7,6 +7,7 @@ class UpgradeManager;
 
 struct UForUpgrade;
 struct ResetGame;
+struct SceneChange;
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Player.h]: Why did u add this?

**Review by jocke1995 — DISMISSED** [review]
Effect looks good, you probably forgot to remove some includes.

File: StortSpelProjekt/Game/src/main.cpp [hunk 3]
```diff
@@ -22,6 +22,8 @@ Scene* GameScene(SceneManager* sm);
 Scene* ShopScene(SceneManager* sm);
 
 void GameInitScene(Scene* scene);
+void ShopInitScene(Scene* scene);
+void ParticleInit();
```
Comment (fjz345) [inline StortSpelProjekt/Game/src/main.cpp:25]: I suppose this function is not meant to be used outside of this class, so private would make more sence.

File: StortSpelProjekt/Game/src/Gamefiles/Player.h (hunk 2 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/Player.h]: Early attempt of solution and unecessary, but fixed now

File: StortSpelProjekt/Game/src/Gamefiles/Player.cpp (hunk 1 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/Player.cpp]: same as below

**Review by jocke1995 — APPROVED** [review]

**Review by fjz345 — APPROVED** [review]

**Pull request merged**